modules/nickserv/register.c

```diff
@@ -83,13 +83,13 @@ ns_cmd_register(struct sourceinfo *si, int parc, char *parv[])
 		return;
 	}
 
-	if (!validemail(email))
```

Comment:
It would be better to indent both conditional blocks and test `! (nicksvs.email_optional && strcasecmp(email, "noemail") == 0)` only once. This would also avoid calling the rather expensive function `email_within_limits()` on garbage data.

You also need to take care of excluding `MU_EMAILMEMOS` from `defuflags` around line 147 if the user has passed `noemail`; otherwise, they will still have `MU_EMAILMEMOS` if the services admin has configured new accounts to have that flag by default.